src/services/transcript-parser.js

```diff
@@ -1006,14 +1006,19 @@ export const markMatchedParts = (text, query, hitCount, hasHighlight = false) =>
     let startIndex = 0;
     let newStr = '';
     for (let j = 0; j < matches.length && count < hitCount;) {
-      if (matches[j] === undefined && matches[j + 3] === undefined) return;
+      if (matches[j] === undefined && matches[j + offset] === undefined) return;
+
+      // Set offset to count matches based on the # of words in the phrase search query
+      const offset = query.split(/[\s-,\?]/)?.length > 0
+        ? (query.split(' ')?.length) * 2 - 1 : 1;
```

Comment:
Is it a problem that `offset` isn't defined until after it is used on line 1009?  Could these two lines be switched?

Reply:
Yes, absolutely!